### PR TITLE
adf4356 driver

### PR DESCRIPTION
**Pull request by sokhramov**
## Pull Request Description

Added driver for adf4356

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
Hello,
Please sign the CLA. Please fix the CI build issues and squash the commits into a single one. (or more if there are multiple unrelated changes only)

**Comment by buha** [comment]
please see this page for how to stack commits if you don't know: https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together

**Review by amiclaus — COMMENTED** [review]
Hello.
Please clean-up the patch. You have changes unrelated to the adf4356 driver in your commits (which i think are already on master branch).

My suggestion would be to have a look at https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History especially on interactive rebase.

File: drivers/frequency/adf5355/adf5355.c [hunk 1]
```diff
@@ -547,4 +554,4 @@ int32_t adf5355_remove(struct adf5355_dev *device)
 	no_os_free(device);
 
 	return ret;
-}
+}
```
Comment (amiclaus) [inline drivers/frequency/adf5355/adf5355.c:557]: please drop this change.

File: drivers/frequency/adf5355/adf5355.h [hunk 2]
```diff
@@ -300,4 +299,4 @@ int32_t adf5355_init(struct adf5355_dev **device,
 		     const struct adf5355_init_param *init_param);
 
 /* Remove the device. */
-int32_t adf5355_remove(struct adf5355_dev *device);
+int32_t adf5355_remove(struct adf5355_dev *device);
```
Comment (amiclaus) [inline drivers/frequency/adf5355/adf5355.h:302]: ditto: drop this change.

File: drivers/frequency/adf5355/adf5355.h [hunk 3]
```diff
@@ -75,18 +75,18 @@
 /* REG5 Bit Definitions */
 #define ADF5355_REG5_DEFAULT			        0x00800025
 
-/* REG6 Bit Definitions */ //CHANGE // INIT for > 75MHz 
-#define ADF5355_REG6_OUTPUT_PWR(x)		        (((x) & 0x3) << 4)
-#define ADF5355_REG6_RF_OUT_EN(x)		        	((x) << 6)
+/* REG6 Bit Definitions */
 #define ADF4355_REG6_OUTPUTB_PWR(x)		        (((x) & 0x3) << 7)
 #define ADF4355_REG6_RF_OUTB_EN(x)		        ((x) << 9)
+#define ADF4356_REG6_RF_OUTB_SEL(x)                ((x) << 25)
```
Comment (amiclaus) [inline drivers/frequency/adf5355/adf5355.h:81]: this looks like an addition. isn't this commit suppose to fix only format issues?

File: drivers/frequency/adf5355/adf5355.c [hunk 4]
```diff
@@ -43,7 +43,7 @@
 #include <inttypes.h>
 #include <stdio.h>
 #include "no_os_error.h"
-#include <stdlib.h>//#include <malloc.h>
+#include <malloc.h>
```
Comment (amiclaus) [inline drivers/frequency/adf5355/adf5355.c:46]: this line is changed in both commits. do it only in one of them please.

**Review by amiclaus — COMMENTED** [review]
please add proper commit subject/body to your commits.

File: drivers/frequency/adf5355/adf5355.c [hunk 5]
```diff
@@ -130,7 +130,8 @@ static int32_t adf5355_reg_config(struct adf5355_dev *dev, bool sync_all)
 	int32_t ret;
 	uint32_t max_reg, i;
 
-	max_reg = ((dev->dev_id == ADF4356) || (dev->dev_id == ADF5356)) ? ADF5355_REG(13) : ADF5355_REG(12);
+	max_reg = ((dev->dev_id == ADF4356)
+		   || (dev->dev_id == ADF5356)) ? ADF5355_REG(13) : ADF5355_REG(12);
```
Comment (rbolboac) [inline drivers/frequency/adf5355/adf5355.c:134]: please perform this change in the first commit

File: drivers/frequency/adf5355/adf5355.c [hunk 6]
```diff
@@ -142,7 +143,7 @@ static int32_t adf5355_reg_config(struct adf5355_dev *dev, bool sync_all)
 		dev->all_synced = true;
 
 	} else {
-		if((dev->dev_id == ADF4356) || (dev->dev_id == ADF5356)){
+		if((dev->dev_id == ADF4356) || (dev->dev_id == ADF5356)) {
```
Comment (rbolboac) [inline drivers/frequency/adf5355/adf5355.c:146]: please perform this change in the first commit

File: drivers/frequency/adf5355/adf5355.c [hunk 7]
```diff
@@ -214,15 +215,16 @@ static int32_t adf5355_set_freq(struct adf5355_dev *dev,
 			freq <<= 1;
 			dev->rf_div_sel++;
 		}
-	} else if (dev->dev_id == ADF4356){
-		if ((freq > dev->max_out_freq) || (freq < dev->min_out_freq) || (!dev->outb_sel_fund))
+	} else if (dev->dev_id == ADF4356) {
+		if ((freq > dev->max_out_freq) || (freq < dev->min_out_freq)
+		    || (!dev->outb_sel_fund))
```
Comment (rbolboac) [inline drivers/frequency/adf5355/adf5355.c:220]: please perform this change in the first commit

File: drivers/frequency/adf5355/adf5355.c [hunk 8]
```diff
@@ -214,15 +215,16 @@ static int32_t adf5355_set_freq(struct adf5355_dev *dev,
...
 			return -EINVAL;
 
 		dev->rf_div_sel = 0;
 
 		dev->regs[ADF5355_REG(6)] = dev->regs[ADF5355_REG(6)] |
 					    ADF4355_REG6_RF_OUTB_EN(dev->outb_en) |
-							ADF4355_REG6_OUTPUTB_PWR(dev->outb_power);
+					    ADF4355_REG6_OUTPUTB_PWR(dev->outb_power);
```
Comment (rbolboac) [inline drivers/frequency/adf5355/adf5355.c:227]: please perform this change in the first commit

File: drivers/frequency/adf5355/adf5355.c [hunk 9]
```diff
@@ -241,7 +243,8 @@ static int32_t adf5355_set_freq(struct adf5355_dev *dev,
 
 	adf5355_pll_fract_n_compute(freq, dev->fpfd, &dev->integer, &dev->fract1,
 				    &dev->fract2, &dev->mod2,
-				    ((dev->dev_id == ADF4356) || (dev->dev_id == ADF5356)) ? ADF5356_MAX_MODULUS2 : ADF5355_MAX_MODULUS2);
+				    ((dev->dev_id == ADF4356)
+				     || (dev->dev_id == ADF5356)) ? ADF5356_MAX_MODULUS2 : ADF5355_MAX_MODULUS2);
```
Comment (rbolboac) [inline drivers/frequency/adf5355/adf5355.c:247]: please perform this change in the first commit

**Review by rbolboac — COMMENTED** [review]
as a rule, the same line of code should be modified in a single commit. in this pull request you do some changes in the first commit and then you style those changes in the second commit. the styling should be applied directly in the first commit when you add the new code.
the 2 commits can be squashed.

**Pull request closed without merging**